addons/stock/models/stock_quant.py

```diff
@@ -134,8 +134,9 @@ def _search_on_hand(self, operator, value):
         """Handle the "on_hand" filter, indirectly calling `_get_domain_locations`."""
         if operator not in ['=', '!='] or not isinstance(value, bool):
             raise UserError(_('Operation not supported'))
-        domain_loc = self.env['product.product']._get_domain_locations()[0]
-        quant_ids = [l['id'] for l in self.env['stock.quant'].search_read(domain_loc, ['id'])]
+        domain_loc = self.env['product.product'].with_context(compute_child=False)._get_domain_locations()[0]
+        location_ids = self.env['stock.location'].search([('id', 'child_of', domain_loc[0][2])]).ids
```

Comment:
get_domain_locations() return a domain. Why not searching on this domain directly instead of `id child_of ... ` ?

Reply:
@Whenrow because the `_get_domain_locations` target normally the quant model not the location one.

The `domain_loc[0][2]` seems a bit tricky. Maybe we should change the ` _get_domain_locations` to return only location_ids.

Reply:
:man_facepalming: Indeed. It's a v14 only patch so we can leave it like that